### PR TITLE
Fix `F.absolute_error` for ChainerX

### PR DESCRIPTION
**Pull request by crcrpar**
Referencing #6071.

### CODE REVIEW

**Comment by niboshi** [comment]
Thanks!
Jenkins, test this please

**Comment by niboshi** [comment]
Oh, it seems `F.absolute_error` needs a fix for ChainerX.
Could you implement `AbsoluteError.forward_chainerx()`?

(Fallback impl `AbsoluteError.forwrad` assigns `self.diff` as `{numpy/cupy}.ndarray`, but that causes array type mismatch in `backward`).

**Comment by niboshi** [comment]
Thanks! Jenkins, test this please

**Comment by niboshi** [comment]
Oh, ChainerX does not support built-in `abs` yet.
How about using `chainerx.abs` instead?

**Comment by crcrpar** [comment]
Soffy for bothering you. Somehow my environment is broken while chainer and chainerx of 7.0.0a1 are installed/imported w/o any error so that I could not find the absence of chainerx's built-in `abs` support.

**Comment by niboshi** [comment]
@crcrpar No problem at all.
Jenkins, test this please.

**Comment by niboshi** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
Please separate `test_backward_non_default_gpu` as another test case.
It does not use any feature of `FunctionTestCase`.
Input sampling is not meant to be done in `setUp`.

File: tests/chainer_tests/functions_tests/loss_tests/test_absolute_error.py [hunk 1]
```diff
@@ -26,74 +26,89 @@
...
+class TestAbsoluteError(testing.FunctionTestCase):
 
     def setUp(self):
-        self.x0 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        if self.dtype == numpy.float16:
+            self.check_forward_options.update({'atol': 1e-3, 'rtol': 1e-3})
+            self.check_backward_options.update({'atol': 5e-2, 'rtol': 5e-2})
+            self.check_double_backward_options.update(
+                {'atol': 5e-2, 'rtol': 5e-2})
+        if self.double_backward_options:
```
Comment (niboshi) [inline tests/chainer_tests/functions_tests/loss_tests/test_absolute_error.py]: This check does not seem reliable.
It's not guaranteed that `self.double_backward_options` is empty by default.

File: tests/chainer_tests/functions_tests/loss_tests/test_absolute_error.py [hunk 2]
```diff
@@ -26,74 +26,89 @@
...
+        # Add sufficient margin to prevent computational error
+        diff = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        diff[abs(diff) < 0.02] = 0.5
+        self.x1 = numpy.asarray(self.x0 + diff)
+        self.gy = numpy.random.random(self.shape).astype(self.dtype)
 
-    # test for #4669
     @attr.multi_gpu(2)
-    def test_backward_non_default_gpu(self):
+    def test_backward_non_default_gpu(self, backend_config):
```
Comment (niboshi) [inline tests/chainer_tests/functions_tests/loss_tests/test_absolute_error.py]: Previously this test did not use `@testing.inject_backend_tests`.
I think there's no strong motivation to run this test for various backends.

**Comment by niboshi** [comment]
Thanks!
Jenkins, test this please

**Comment by niboshi** [comment]
I changed the title of the PR because it contains a bug fix for ChainerX.

**Pull request merged**